tests/handlers/aws/test_outputs.py

```diff
@@ -0,0 +1,135 @@
+    _logs_retrieve_event_from_cloudwatch_logs,
+    _logs_upload_event_to_cloudwatch_logs,
+    _s3_upload_content_to_bucket,
+)
+from tests.testcontainers.logstash import LogstashContainer
+
+
+@pytest.mark.integration
+class TestLambdaHandlerLogstashOutputSuccess(TestCase):
+    def setUp(self) -> None:
+        lst = LocalStackContainer(image="localstack/localstack:1.1.0")
+        lst.with_env("EAGER_SERVICE_LOADING", "1")
+        lst.with_services("s3", "logs", "sqs")
+        self.localstack = lst.start()
+
+        aws_default_region = "us-east-1"
+        session = boto3.Session(region_name=aws_default_region)
+        self.s3_client = session.client("s3", endpoint_url=self.localstack.get_url())
+        self.logs_client = session.client("logs", endpoint_url=self.localstack.get_url())
+        self.sqs_client = session.client("sqs", endpoint_url=self.localstack.get_url())
+
+        self.logstash_http_port = 5043
+        lgc = LogstashContainer(port=self.logstash_http_port)
+        logstash_config = """\
+            input {{
+              http {{
+                port => {logstash_http_port}
+                codec => json_lines
+              }}
+            }}
+
+            output {{ stdout {{ codec => json_lines }} }}
+            """.format(
+            logstash_http_port=self.logstash_http_port
+        )
+        print(logstash_config)
+        lgc.with_env("CONFIG_STRING", logstash_config)
+        self.logstash = lgc.start()
+
+        self.fixtures = {
+            "cw_log_1": _load_file_fixture("cloudwatch-log-1.json"),
+            "cw_log_2": _load_file_fixture("cloudwatch-log-2.json"),
+        }
+
+        group_name = _class_based_id(self, suffix="source-group")
+        stream_name = _class_based_id(self, suffix="source-stream")
+
+        _logs_create_cloudwatch_logs_group(self.logs_client, group_name=group_name)
+        g = _logs_create_cloudwatch_logs_stream(self.logs_client, group_name=group_name, stream_name=stream_name)
+        cloudwatch_group_arn = g["arn"]
+
+        _logs_upload_event_to_cloudwatch_logs(
+            self.logs_client,
+            group_name=group_name,
+            stream_name=stream_name,
+            messages_body=[self.fixtures["cw_log_1"] + self.fixtures["cw_log_2"]],
+        )
+
+        self.group_name = group_name
+        self.stream_name = stream_name
+
+        config_content = _load_file_fixture("config.yaml")
+        self.config = Template(config_content).substitute(
+            dict(CloudwatchLogStreamARN=cloudwatch_group_arn, LogstashURL=self.logstash.get_url())
+        )
+        print(self.config)
+
+        _s3_upload_content_to_bucket(
+            client=self.s3_client,
+            content=self.config,
+            content_type="text/plain",
+            bucket_name=_class_based_id(self, suffix="config-bucket").lower(),
+            key="folder/config.yaml",
+        )
+
+        def _create_sqs_queue(client: BotoBaseClient, name: str) -> Any:
```

Comment:
any reason for having this function defined in the `setUp` instead that as an helper?

Reply:
Forgot about moving it, will do